nengo/dists.py

```diff
@@ -306,6 +346,358 @@ def sample(self, n, d=None, rng=np.random):
+        """Map samples from the ``[0, 1]``-cube onto the hypersphere.
+
+        Uses the SCT method described in section 1.5.3 of Fang and Wang (1994).
+        """
+        samples = np.asarray(samples)
+        samples = samples[:, np.newaxis] if samples.ndim == 1 else samples
+        n, d = samples.shape
+
+        # inverse transform method (section 1.5.2)
+        coords = np.empty_like(samples)
+        for j in range(d):
+            coords[:, j] = cls.spherical_coords_ppf(d - j, samples[:, j], approx=approx)
+
+        # spherical coordinate transform
+        mapped = np.ones((n, d + 1))
+        i = np.ones(d)
+        i[-1] = 2.0
+        s = np.sin(i[np.newaxis, :] * np.pi * coords)
+        c = np.cos(i[np.newaxis, :] * np.pi * coords)
+        mapped[:, 1:] = np.cumprod(s, axis=1)
+        mapped[:, :-1] *= c
+        return mapped
+
+    @staticmethod
+    def spherical_transform_tfww(c_samples):
+        """Map samples from the ``[0, 1]``-cube onto the hypersphere surface.
+
+        Uses the TFWW method described in section 4.3 of Fang and Wang (1994).
+        """
+        c_samples = np.asarray(c_samples)
+        c_samples = c_samples[:, np.newaxis] if c_samples.ndim == 1 else c_samples
+        n, s1 = c_samples.shape
+        s = s1 + 1
+
+        x_samples = np.zeros((n, s))
+
+        if s == 2:
+            phi = 2 * np.pi * c_samples[:, 0]
+            x_samples[:, 0] = np.cos(phi)
+            x_samples[:, 1] = np.sin(phi)
+            return x_samples
+
+        even = s % 2 == 0
+        m = s // 2 if even else (s - 1) // 2
+
+        g = np.zeros((n, m + 1))
+        g[:, -1] = 1
+        for j in range(m - 1, 0, -1):
+            g[:, j] = g[:, j + 1] * c_samples[:, j - 1] ** (
+                (1.0 / j) if even else (2.0 / (2 * j + 1))
+            )
+
+        d = np.sqrt(np.diff(g, axis=1))
+
+        phi = c_samples[:, m - 1 :]
+        if even:
+            phi *= 2 * np.pi
+            x_samples[:, 0::2] = d * np.cos(phi)
+            x_samples[:, 1::2] = d * np.sin(phi)
+        else:
+            # there is a mistake in eq. 4.3.7 here, see eq. 1.5.28 for correct version
+            phi[:, 1:] *= 2 * np.pi
+            f = 2 * d[:, 0] * np.sqrt(phi[:, 0] * (1 - phi[:, 0]))
+            x_samples[:, 0] = d[:, 0] * (1 - 2 * phi[:, 0])
+            x_samples[:, 1] = f * np.cos(phi[:, 1])
+            x_samples[:, 2] = f * np.sin(phi[:, 1])
+            if s > 3:
+                x_samples[:, 3::2] = d[:, 1:] * np.cos(phi[:, 2:])
+                x_samples[:, 4::2] = d[:, 1:] * np.sin(phi[:, 2:])
+
+        return x_samples
+
+    @staticmethod
+    def random_orthogonal(d, rng=np.random):
+        """Returns a random orthogonal matrix."""
+        m = rng.standard_normal((d, d))
+        u, _, v = np.linalg.svd(m)
+        return np.dot(u, v)
+
+    def sample(self, n, d=1, rng=np.random):
+        if d == 1:
+            return super().sample(n, d, rng=rng)
+
+        base = QuasirandomSequence()
```

Comment:
I might put this in the constructor, partly for performance and partly so that if someone (probably one of us) wants to play around and try a different base, we can. I'm fine keeping it hidden (e.g. in `self._base`).

Reply:
Ended up reverting this part of things since there are other sequences that could be put in eventually.